drivers/flash/nrf_qspi_nor.c

```diff
@@ -293,6 +325,34 @@ static inline void qspi_unlock(const struct device *dev)
 #endif
 }
 
+/* Everything necessary to acquire owning access to the device.
+ *
+ * This means taking the lock and, if necessary, waking the device
+ * from deep power-down mode.
+ */
+static inline void qspi_acquire(const struct device *dev)
+{
+	qspi_lock(dev);
+
+#ifdef CONFIG_NORDIC_QSPI_NOR_IDLE_IN_DPD
+	exit_dpd(dev, true);
+#endif
+}
```

Comment:
this is kind of mimicking the `pm_device_runtime` API, so I'm not happy to see alternatives being used. Any reason why it's not used?

Reply:
Discussion here: https://github.com/zephyrproject-rtos/zephyr/pull/42052#discussion_r790806878

Essentially: This provides further benefits, and easier implementation than pm_device_runtime, by shutting down the device when it's not actively in use, and automatically controlling device power state, rather than depending on the application to enter/exit standby for the device before and after every call that might result in flash access

This behavior mimics the IDLE_IN_DPD support for the generic SPI NOR flash implementation

Reply:
> Essentially: This provides further benefits, and easier implementation than pm_device_runtime, by shutting down the device when it's not actively in use, and automatically controlling device power state, rather than depending on the application to enter/exit standby for the device before and after every call that might result in flash access
> 

this is precisely what the device runtime PM API does.

Reply:
I was previously just looking at the CONFIG_PM_DEVICE, but not CONFIG_PM_DEVICE_RUNTIME - It looks like we could just replace the enter_dpd and exit_dpd calls within aquire/release with pm_device_runtime_get / pm_device_runtime_put?

I didn't see any examples of this in the various flash drivers, but it does seem like it should work, I'll try to get something worked up and tested today or tomorrow.

Reply:
Sorry for the delays on this, I got pulled off onto another project for a little bit, still will be returning to this.

Reply:
Updated to use PM_DEVICE_RUNTIME.